### PR TITLE
Fix miscellaneous command permission bugs

### LINKED ISSUE #3612: /list command exposes vanished groups
### Information

**Full output of `/ess version`**:

```
Server version: 1.15.2-R0.1-SNAPSHOT git-Paper-385 (MC: 1.15.2)
EssentialsX version: 2.18.0.0
PlaceholderAPI version: 2.10.9
LuckPerms version: 5.1.93
Vault version: 1.7.3-b131
Citizens version: 2.0.27-SNAPSHOT (build 2072)
EssentialsXChat version: 2.18.0.0
EssentialsXSpawn version: 2.18.0.0
```

**Server log**: there's no errors in the console - not relevant

**EssentialsX config**: https://gist.github.com/piotrskibapl/901bde367422ed7422ac6952f71a609b

### Details

**Description**  
If a group has only vanished members online, it's still in the "/list" tab completions even if a player can't see any players that belong to this group.

**Steps to reproduce**  
1. Log in on the server
2. Make sure that there's at least one member of a certain group online, and all of the online group members are vanished and you can't see them
3. Type in "/list " and press tab
4. The list of groups that have any online members is displayed in the tab completions. It also contains the group that has only vanished players online.

**Expected behavior**  
A group shouldn't be exposed in the tab completions if you can't see any members of it.

### LINKED ISSUE #3613: Permissions for changing time in world not being checked.
### Information

**Full output of `/ess version`**:

```
CONSOLE issued server command: /ess version
Server version: 1.16.1-R0.1-SNAPSHOT git-Paper-137 (MC: 1.16.1)
EssentialsX version: 2.18.0.27
LuckPerms version: 5.1.93
Vault version: 1.7.3-b131
You are running an unsupported server version!
```

**Server log**: https://gist.github.com/Fefo6644/3c82439498e12fa59890937b82df7724

**EssentialsX config**: https://gist.github.com/Fefo6644/d56e7bc7527778b789195825d35bbe30

### Details

**Description**  
The permission for setting time (`essentials.time.set`) is not being checked when running commands to change time.

**Steps to reproduce**  
1_ Set `essentials.time` to true, either directly to the user or in an inherited group.
2_ Set `essentials.time.set` to false, either directly to the user or in an inherited group.
3_ Run any command to change world time, such as `/day`, `/night` or any variation of `/time set <time>`

**Expected behavior**  
The action shouldn't be allowed and the world time shouldn't change.

**Screenshots**

### PR DESCRIPTION
**Pull request by JRoy**
some of these bugs aren't mine and have nothing to do with command refactor pr but I fixed them anyway

Fix #3612
Fix #3613

### CODE REVIEW

File: Essentials/src/com/earth2me/essentials/commands/Commandtime.java [hunk 1]
```diff
@@ -64,14 +64,16 @@ public void run(final Server server, final CommandSource sender, final String co
...
+
+        final StringJoiner joiner = new StringJoiner(",");
         for (World world : worlds) {
             if (!canUpdateWorld(user, world)) {
                 throw new Exception(tl("timeSetWorldPermission", user.getWorld().getName()));
             }
-        }
 
-        final StringJoiner joiner = new StringJoiner(",");
-        for (World world : worlds) {
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/commands/Commandtime.java]: Merging these loops almost certainly result in incorrect behavior, as the command will be able to fail while also being able to partially update the requested worlds, and then also not providing feedback to the user for what worlds were changed.

**Review by pop4959 — CHANGES_REQUESTED** [review]

File: Essentials/src/com/earth2me/essentials/commands/Commandtime.java (hunk 1 above)
Comment (JRoy) [reply Essentials/src/com/earth2me/essentials/commands/Commandtime.java]: that explains why i didn't do it, lemme fix

**Review by pop4959 — DISMISSED** [review]

File: Essentials/src/com/earth2me/essentials/commands/EssentialsCommand.java [hunk 2]
```diff
@@ -285,9 +285,9 @@ private static boolean canInteractWith(User interactor, User interactee) {
     /**
      * Returns a list of all online groups.
      */
-    protected List<String> getGroups() {
+    protected List<String> getGroups(CommandSource interactor) {
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/commands/EssentialsCommand.java]: This is only used in `/list`; we could probably just move this to `Commandlist`.

Comment (JRoy) [reply Essentials/src/com/earth2me/essentials/commands/EssentialsCommand.java]: cool and good

**Review by pop4959 — APPROVED** [review]

**Review by mdcfe — APPROVED** [review]

**Pull request merged**